sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -353,7 +370,58 @@ struct graphs_support;
 } // namespace device
 } // namespace info
 
-class node {};
+enum class dynamic_parameter_type {
+  usm,
+  buffer,
+  scalar
+};
+
+template <dynamic_parameter_type DynParamType>
+class dynamic_parameter {
+public:
+  dynamic_parameter(std::string label = "");
+
+  const std::string& get_label() const;
+};
+
+template <>
+class dynamic_parameter<dynamic_parameter_type::scalar> {
+public:
+  dynamic_parameter(size_t size, std::string label = "");
+
+  size_t get_size() const;
+  const std::string& get_label() const;
+};
+
+class node {
+public:
+  node() = delete;
+
+  /* -- parameter registration interface -- */
+
+  void register_dynamic_parameter(void* param,
+                                  dynamic_parameter<dynamic_parameter_type::usm> dynamicParam) const;
```

Comment:
which one is updated by calling the function register_dynamic_parameter?  "this" is 
const, "void* param" is the input, and so the reference "&" is missed for dynamicParam?

Reply:
It's `dynamicParam` that is modified, that might be clearer if we make register a method on the `dynamic_parameter` object and pass in the `node` object. I'll try draft that up and see how it looks.

I was assuming that the `dynamic_parameter` would have the same Command Reference Semantics as other sycl objects and so be reference counted. I've not specified that however, and will do.

Reply:
> the dynamic_parameter would have the same Command Reference Semantics as other sycl objects and so be reference counted

thanks, with this in mind, I think the current API is a good design.  Anyway, I'm open to what method is used.